spec/beaker-puppet/install_utils/foss_utils_spec.rb

```diff
@@ -1353,6 +1347,23 @@ def test_fetch_http_file_no_ending_slash(platform)
     it 'calls fetch_http_file with no ending slash' do
       test_fetch_http_file_no_ending_slash( 'debian-5-x86_64' )
     end
+
+    it 'sets correct file paths for el6' do
+      platform = Object.new()
+      allow(platform).to receive(:to_array) { ['el', '6', 'x4'] }
+      host = basic_hosts.first
+      host['platform'] = platform
+      opts = { :puppet_agent_version => 'agentversion' , :pe_promoted_builds_url => 'http://pm.puppetlabs.com'}
+      collection = 'puppet1'
+
+      allow( subject ).to receive( :options ).and_return( opts )
+      allow( subject ).to receive( :scp_to )
+      allow( subject ).to receive( :configure_type_defaults_on ).with(host)
+      expect( subject ).to receive( :get_puppet_collection).with('agentversion').and_return(collection)
```

Comment:
Beaker is horrible to unit test, since the method is all side effects, but it shouldn't be necessary to mock get_puppet_collection(), that's part of what we are testing after all, and it's just a function.

Reply:
Hmm, that would cause nothing to be returned from get_puppet_collection. I was suggesting that it not be mocked it at all.

Reply:
That's my mistake, I thought that spec was for the correct invocation of method `#get_puppet_collection`, not for the whole calling method. This does seem to be a lot of mocking for the method under consideration.